graphql/schema/types/project_settings.graphql

```diff
@@ -20,6 +20,7 @@ It contains information about project settings (e.g. Build Baron configurations,
 update the settings for a given project.
 """
 input ProjectSettingsInput {
+  projectId: String # TODO: Make this required.
```

Comment:
I'm not sure how I feel about duplicating the same field twice (here and in `projectRef.Id`, and in other structs). What do you think is the main benefit?

Reply:
Ahh the reason I did this is if we have `projectRef.Id` or `repo.Id`, just having `Id` isn't specific enough to describe what type of input it is. I think we wouldn't be able to put the new directive on the `projectRef` or `repoRef` objects because they wouldn't convert properly into a map, and we'd have to put it on the `Id` field. I'm open to other ideas though!

Reply:
Ah the new directives, that makes perfect sense! Thanks 🤩

Reply:
In https://github.com/evergreen-ci/evergreen/pull/7644 we discussed using projectIdentifier as the key for projectIds do we still want to do that?

Reply:
I'm sort of against it in this case because this input mirrors repos, and the repo uses `ID`

Reply:
Makes sense thanks for clarifying!